File: Services/TaskService/TaskService.Application/UseCases/TaskUseCases/UpdateTask/UpdateTaskCommand.cs
using MediatR;

namespace TaskService.Application.UseCases
{
    public record UpdateTaskCommand(
        string? username,
        int? companyId,
        int? taskId,
        string? name,
        string? description,
        decimal? budget,
        Domain.Enums.TaskStatus? status
    ) : IRequest<Unit>;
}

File: FrontendApp/src/app/models/commands/taskservice/update-task.command.ts
import { TaskStatus } from "../../../enums/task-status.enum";

export interface UpdateTaskCommand {
  username: string | null;
  companyId: number | null;
  taskId: number | null;
  name: string | null;
  description: string | null;
  budget: number | null;
  status: TaskStatus | null;
}
